specs/_features/maxeb_increase/capella.py

```diff
@@ -1627,6 +1652,83 @@ def process_pending_balance_deposits(state: BeaconState) -> None:
     state.pending_balance_deposits = state.pending_balance_deposits[next_pending_deposit_index:]
 
 
+def apply_pending_consolidation(state: BeaconState, pending_consolidation: PendingConsolidation) -> None:
+    current_epoch = get_current_epoch(state)
+
+    # Resolve the target of consolidation
+    target_index = resolve_consolidated_to(state, pending_consolidation.target_index)
+    target_validator = state.validators[target_index]
+    source_validator = state.validators[pending_consolidation.source_index]
+
+    # Apply consolidation
+    source_validator.consolidated_to = target_index
+
+    # Don't move the balance and slash the target if the source was slashed
```

Comment:
When the target is slashed, we do need to slash the source as well, to prevent the case where consolidation is aborted on one chain but not on another, and the source balance manages to double vote without consequences on the former (on the chain where the consolidation was aborted, the source validator votes one way, and on chain where the consolidation happened the target validator votes another way. Slashing of the source balance would only happen on the latter).
But why do we need to slash the target if the source has been slashed? I don't see anything wrong with just aborting in this case, since the source balance has anyway been slashed, and that's the only balance which might try to avoid slashing through the consolidation

Reply:
It can be the case that finalizing a consolidation and including slashing the source on chain have different order on different forks which would create an incentives imbalance. As if the target is slashed on one fork but not on another, it could create an incentive for a party owning the target to stop voting on the fork where the target appeared to be slashed.